simulator/func_sim/alu.h

```diff
@@ -230,6 +302,19 @@ struct ALU
     // Bit permutation
     template<typename I> static void grev( I* instr) { instr->v_dst = gen_reverse( instr->v_src1, shamt_v_src2<typename I::RegisterUInt>( instr)); }
 
+    template<typename I> static
+    void riscv_unshfl( I* instr)
+    {
+        auto dst_value = instr->v_src1;
+        constexpr size_t limit = 4 + bytewidth<decltype( instr->v_src1)> / bytewidth<uint64>;
```

Comment:
I believe you need this minus 1:

https://github.com/MIPT-ILab/mipt-mips/blob/41aa4aa9f166a59c8c8f61137623327328e60f60/simulator/infra/t/unit_test.cpp#L118-L119